rpc-tests/tests/test-dapp-staking.js

```diff
@@ -0,0 +1,27 @@
+import { expect } from 'chai';
+import { Keyring } from '@polkadot/api';
+import { describeWithAstar, wait } from './util.js';
+
+const CONTRACT = '0x000000000000000000000000000000000000000001'; //0x01
+
+describeWithAstar('Dapp Staking', function(context) {
```

Comment:
Good start!

However, the scenario described in the Notion ticket isn't covered here.